api/routes/update.ts

```diff
@@ -387,6 +395,19 @@ async function recordSearch(
   destinyVersion: DestinyVersion,
   payload: UsedSearchUpdate['payload']
 ): Promise<ProfileUpdateResult> {
+  // TODO: I did a silly thing and made the query part of the search table's
+  // primary key, instead of using a fixed-size hash of the query. This limits
+  // how big a query we can store (and bloats the index). I'll need to do some
+  // surgery to fix that, but we should probably just generally refuse to save
+  // long queries.
+  if (payload.query.length > 2048) {
+    metrics.increment('update.validation.searchTooLong.count');
+    return {
+      status: 'InvalidArgument',
```

Comment:
will this cause an error message somewhere it's being consumed? should it? seems low stakes

Reply:
It won't, as of now. DIM silently swallows these.